main.m

```diff
@@ -7,8 +7,57 @@
 //
 
 #import <Cocoa/Cocoa.h>
+#import "Extractor.h"
+
+/** Check for input filename in CLI args, if any.
+ *  Input arg qualifies as either:
+ *      - arg at index 1 that does not start with "-"
+ *      - arg directly after an arg that matches exactly "-i"
+ *  Return index of filename, or 0 on failure
+ */
+char * findInputArg(int argc, char *argv[]) {
+    if (argc < 2) { return nil; }
+    if (argv[1][0]!='-') { return argv[1]; }
+    for (int i=1; i<argc-1; i++) {
+        if (strcmp(argv[i], "-i")==0) {
+            return argv[i+1];
+        }
+    }
+    return nil;
+}
+
+/** Check for out dirname in CLI args, if any.
+ *  Must be:
+ *      - arg directly after an arg that matches exactly "-o"
+ *  Return indx of dirname, or 0 on failure
+ */
+char * findOutputArg(int argc, char *argv[]) {
+    if (argc < 3) { return nil; }
+    for (int i=1; i<argc-1; i++) {
+        if (strcmp(argv[i], "-o")==0) {
+            return argv[i+1];
+        }
+    }
+    return nil;
+}
```

Comment:
This is cool, and looks like good fun :) but did you try getopt? https://pubs.opengroup.org/onlinepubs/000095399/functions/getopt.html has a few safety bits that might be nice.

Reply:
No, it isn't intuitive at all :) it just covers some edge cases, and saves you from having to reimplement a few things. It does fall short in some areas too, but it is a useful tool to have at your disposal 🛠️